pkgs/tools/networking/xrootd/default.nix

```diff
@@ -68,10 +71,31 @@ stdenv.mkDerivation rec {
     sed -i 's/set\((\s*CMAKE_INSTALL_[A-Z_]\+DIR\s\+"[^"]\+"\s*)\)/define_default\1/g' cmake/XRootDOSDefs.cmake
   '';
 
+  # https://github.com/xrootd/xrootd/blob/master/packaging/rhel/xrootd.spec.in#L665-L675=
+  postInstall = ''
+    mkdir -p "$out/lib/tmpfiles.d"
+    install -m 644 -T ../packaging/rhel/xrootd.tmpfiles "$out/lib/tmpfiles.d/xrootd.conf"
+    mkdir -p "$out/etc/xrootd"
+    install -m 644 -t "$out/etc/xrootd" ../packaging/common/*.cfg
+    install -m 644 -t "$out/etc/xrootd" ../packaging/common/client.conf
+    mkdir -p "$out/etc/xrootd/client.plugins.d"
+    install -m 644 -t "$out/etc/xrootd/client.plugins.d" ../packaging/common/client-plugin.conf.example
+    mkdir -p "$out/etc/logrotate.d"
+    install -m 644 -T ../packaging/common/xrootd.logrotate "$out/etc/logrotate.d/xrootd"
+  '' + lib.optionalString stdenv.isLinux ''
+    mkdir -p "$out/lib/systemd/system"
+    install -m 644 -t "$out/lib/systemd/system" ../packaging/common/*.service ../packaging/common/*.socket
+  '';
+
   cmakeFlags = lib.optionals enableTests [
     "-DENABLE_TESTS=TRUE"
   ];
 
+  postFixup = lib.optionalString (externalEtc != null) ''
+    mv "$out"/etc{,.orig}
+    ln -s ${lib.escapeShellArg externalEtc} "$out/etc"
```

Comment:
Can we set the `CMAKE_INSTALL_SYSCONFDIR` instead?

Reply:
Yes, since the `"$out/etc"` contents are managed manually (by the postInstall commands), not by the build tools.

However, if one day the upstream decides to put something to /etc through `make install`, we'll have to alter the Nix expression to keep the SYSCONFDIR as `"$out/etc"` and move `$out/etc` to `$out/etc.orig` afterward. That would also change the directory structure of the output.